core/system/src/mpu/vmpu_freescale_k64_mem.c

```diff
@@ -33,19 +57,61 @@ typedef struct
     uint32_t count;
 } TBoxACL;
 
-uint32_t g_mem_acl_count, g_mem_acl_user;
+uint32_t g_mem_acl_count, g_mpu_slot;
 static TMemACL g_mem_acl[UVISOR_MAX_ACLS];
 static TBoxACL g_mem_box[UVISOR_MAX_BOXES];
 
+int vmpu_mem_push_page_acl(uint32_t start_addr, uint32_t end_addr)
+{
+    TMemACL rgd;
+    volatile uint32_t* word;
+
+    /* Check that start and end address are aligned to 32-byte. */
+    if (start_addr & 0x1F || end_addr & 0x1F) {
+        return -1;
+    }
+
+    rgd.word[0] = start_addr;
+    rgd.word[1] = end_addr;
+    /* Permissions are fixed to only allow unprivileged write/read/execute. Duh. */
+    rgd.word[2] = 0x1E;
+
+    /* Insert into MPU regions in RR fashion. */
+    word = MPU->WORD[g_mpu_slot];
```

Comment:
@niklas-arm Just to be sure, please check the original KSDK files from Freescale, both in the old version and the new one (KSDK 2).

The file `mpu_kinetis.h` is a reduced version of the original one that only defines MPU-specific symbols. Some of them might have been omitted by mistake (from the original file in the KSDK1), or maybe the new KSDK2 now defines them.